### PR TITLE
feat: allow to use 'six.moves.collections_abc.Mapping' in 'client_options.from_dict()'

### PR DESCRIPTION
**Pull request by paul1319**

### CODE REVIEW

**Review by IlyaFaer — APPROVED** [review]
Looks like the things Bu Sun Kim was talking about.

**Comment by IlyaFaer** [comment]
@paul1319, I think, we could add a unit test with other mapping object to be sure

**Review by IlyaFaer — APPROVED** [review]

File: googleapiclient/discovery.py [hunk 1]
```diff
@@ -204,8 +204,9 @@ def build(
     cache_discovery: Boolean, whether or not to cache the discovery doc.
     cache: googleapiclient.discovery_cache.base.CacheBase, an optional
       cache object for the discovery documents.
-    client_options: Dictionary or google.api_core.client_options, Client options to set user
-      options on the client. API endpoint should be set through client_options.
+    client_options: Mapping object or google.api_core.client_options, Client
+      options to set user options on the client. API endpoint should be set
```
Comment (mf2199) [inline googleapiclient/discovery.py]: `client` should be written in the same case, upper or lower. Also, `The API endpoint...`

File: tests/test_discovery.py [hunk 2]
```diff
@@ -534,6 +534,19 @@ def test_api_endpoint_override_from_client_options_dict(self):
 
         self.assertEqual(plus._baseUrl, api_endpoint)
 
+    def test_api_endpoint_override_from_client_options_mapping_object(self):
+        from collections import defaultdict
```
Comment (mf2199) [inline tests/test_discovery.py]: This can be placed at the top of the file, alongside the other imports.

**Review by mf2199 — CHANGES_REQUESTED** [review]

**Review by IlyaFaer — COMMENTED** [review]
`The` should be added into both statements `API endpoint should be set through client_options` (though I don't think it actually should, @mf2199, WDYS?)

**Comment by mf2199** [comment]
> `The` should be added into both statements `API endpoint should be set through client_options` (though I don't think it actually should, @mf2199, WDYS?)

:+1: There are two such statements one of which contains the article; the second can be copied for the first.

**Pull request closed without merging**